server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -680,6 +681,14 @@ public void writeMapWithConsistentOrder(@Nullable Map<String, ? extends Object>
             o.writeString(zonedDateTime.getZone().getId());
             o.writeLong(zonedDateTime.toInstant().toEpochMilli());
         });
+        writers.put(JodaCompatibleZonedDateTime.class, (o, v) -> {
+            // write the joda compatibility datetime as joda datetime
+            o.writeByte((byte) 13);
+            final JodaCompatibleZonedDateTime zonedDateTime = (JodaCompatibleZonedDateTime) v;
+            String zoneId = zonedDateTime.getZonedDateTime().getZone().getId();
```

Comment:
`zoneId` is unused here?

Reply:
Oops. This was because the zoneId used by joda doesn't appear to always be the same as java time. Joda does not know how to parse "Z". I'll add special casing for that.

Reply:
I pushed f105096